### PR TITLE
Add support for checking UUCP files.

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

**Comment by nomis52** [comment]
Closes #716

File: plugins/usbpro/UsbSerialPlugin.cpp [hunk 1]
```diff
@@ -305,6 +313,14 @@ bool UsbSerialPlugin::SetDefaultPreferences() {
     save = true;
   }
 
+  vector<string> lock_paths =
+    m_preferences->GetMultipleValue(UUCP_LOCK_PATH_KEY);
+  if (lock_paths.empty()) {
+    m_preferences->SetMultipleValue(UUCP_LOCK_PATH_KEY, UUCP_MAC_PATH);
+    m_preferences->SetMultipleValue(UUCP_LOCK_PATH_KEY, UUCP_LINUX_PATH);
+    save |= true;
```
Comment (peternewman) [inline plugins/usbpro/UsbSerialPlugin.cpp]: Just save = true surely?

File: plugins/usbpro/WidgetDetectorThread.cpp [hunk 2]
```diff
@@ -207,6 +217,12 @@ bool WidgetDetectorThread::RunScan() {
     if (StringEndsWith(*it, ".init") || StringEndsWith(*it, ".lock"))
       continue;
 
+    const string base_name = ola::file::FilenameFromPath(*it);
+    if (!base_name.empty() &&
+        ola::io::CheckForUUCPLockFile(m_uucp_lock_paths, base_name)) {
+      OLA_INFO << "Locked USB Service device at " << *it;
```
Comment (peternewman) [inline plugins/usbpro/WidgetDetectorThread.cpp]: SPaG Serial?

Comment (peternewman) [inline plugins/usbpro/WidgetDetectorThread.cpp]: Are we not going to skip discovering it if it's locked?

File: plugins/usbpro/UsbSerialPlugin.cpp (hunk 1 above)
Comment (nomis52) [inline plugins/usbpro/UsbSerialPlugin.cpp]: Done

**Pull request merged**